### PR TITLE
Dispute Detail: Display error on invalid dispute

### LINKED ISSUE #180: Disputes: invalid dispute id in URL results in always loading dispute
E.g. see https://court-rinkeby.aragon.org/disputes/100

The dispute detail ends up in an always "loading" state, but we should explain to the user that this dispute is not available.

### PR DESCRIPTION
**Pull request by fabriziovigevani**
closes #180 

TODO: 

- [x] Merge new #203 changes

### CODE REVIEW

File: src/components/Disputes/DisputeDetail.js [hunk 1]
```diff
@@ -79,43 +69,52 @@ const DisputeDetail = React.memo(function DisputeDetail({ match }) {
...
-                ) : (
-                  <DisputeTimeline dispute={dispute} />
-                )}
-              </Box>
-            </React.Fragment>
-          }
+      {noDispute ? (
+        <ErrorLoading
+          subject="dispute"
+          errors={[`Dispute #${disputeId} does not exist`]}
```
Comment (rperez89) [inline src/components/Disputes/DisputeDetail.js]: should we check this error msg with @dizzypaty ?

Comment (dizzypaty) [reply src/components/Disputes/DisputeDetail.js]: > should we check this error msg with @dizzypaty ?

@rperez89 – See designs for this error below: 

dispute-not-found

Figma file: ✅ MASTER: Court Dashboard > Page: - Error pages
https://www.figma.com/proto/IdVA6iMn0cWLV5imylwEUP/%E2%9C%85-MASTER-Court-Dashboard?node-id=4158%3A52&viewport=170%2C1002%2C0.25&scaling=min-zoom

File: src/GlobalErrorHandler.js [hunk 2]
```diff
@@ -55,11 +57,15 @@ class GlobalErrorHandler extends React.Component {
 
     return error ? (
       <GlobalErrorScreen>
-        <GenericError
-          detailsTitle={error.message}
-          detailsContent={errorStack}
-          reportCallback={SENTRY_DSN ? this.handleReportClick : null}
-        />
+        {error instanceof DisputeNotFound ? (
```
Comment (rperez89) [inline src/GlobalErrorHandler.js]: Not sure if we should do this, this is the global error the dispute not found is not a global error is an specific error and should not be handled by the global error which intention is to capture all the errors that were not captured by any specific component error. So here i would suggest to re use the `GlobalErrorScreen` and pass a children with the specific error. But not handle the dispute error inside the global error

File: src/hooks/useDisputes.js [hunk 3]
```diff
@@ -55,10 +55,35 @@ export default function useDisputes() {
...
+    ? convertToString(Object.values(disputePhase)[0])
+    : ''
+
+  return useMemo(() => {
+    if (fetching || (dispute && !disputeProcessed)) {
+      return { fetching: true }
+    }
+
+    return {
+      dispute: dispute
```
Comment (rperez89) [inline src/hooks/useDisputes.js]: should we use a function here to avoid returning with inline ternary operation?

File: src/hooks/useJurorSubscriptionFees.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from 'react'
```
Comment (rperez89) [inline src/hooks/useJurorSubscriptionFees.js]: i think that this file does not belong to this pr right?

File: src/GlobalErrorHandler.js (hunk 2 above)
Comment (fabriziovigevani) [reply src/GlobalErrorHandler.js]: Not sure i follow. I think that this component is not meant to only capture unhandled generic errors but also any error that we would like to handle (i take that by `Global` doesn't reference a generic error but the error hub of the app). Doesn't make sense to me to pass another component as children that will use the `GlobalErrorScreen` when we are already using it here.

File: src/hooks/useDisputes.js (hunk 3 above)
Comment (fabriziovigevani) [reply src/hooks/useDisputes.js]: I don't think this is a ternary operation?

File: src/hooks/useJurorSubscriptionFees.js (hunk 4 above)
Comment (fabriziovigevani) [reply src/hooks/useJurorSubscriptionFees.js]: Oops yeah definitely

File: src/GlobalErrorHandler.js (hunk 2 above)
Comment (rperez89) [reply src/GlobalErrorHandler.js]: you don't need the globalErrorHandler at all this is a dispute detail error the same than any of the dispute detail error, think about if not we are going to have some dispute errors here another dispute details error inside the dispute detail component, what you need is the `<GlobalErrorScreen/>` not the handler, also i think that you created the `disputeNotFound` inside a generic error folder instead inside the `Dispute` Folder as the other dispute errors

File: src/hooks/useDisputes.js (hunk 3 above)
Comment (rperez89) [reply src/hooks/useDisputes.js]: `? ` `:` is the ternary operator

File: src/GlobalErrorHandler.js (hunk 2 above)
Comment (rperez89) [reply src/GlobalErrorHandler.js]: take a look [here](https://reactjs.org/docs/error-boundaries.html) if you want to what is the idea for this component

Comment (fabriziovigevani) [reply src/GlobalErrorHandler.js]: I think we do need the global error handler. We need to treat this error different because we are going to display it in a [different way](https://github.com/aragon/court-dashboard/pull/198#discussion_r385010775) than the other types of dispute errors that are local to the dispute detail. Not sure if you are suggesting to use  `<GlobalErrorScreen />`  inside the dispute detail, but that would not work because it would render as a child of `<MainView />`.

File: src/hooks/useDisputes.js (hunk 3 above)
Comment (fabriziovigevani) [reply src/hooks/useDisputes.js]: Ah thought you meant nested operations. What's the problem to use it like this? Don't see a benefit of using a function for this simple operation.

File: src/GlobalErrorHandler.js (hunk 2 above)
Comment (rperez89) [reply src/GlobalErrorHandler.js]: ohh is a full screen error this one! so yes you are right, so let's change the dispute error inside the dispute component folder maybe?

File: src/hooks/useDisputes.js (hunk 3 above)
Comment (rperez89) [reply src/hooks/useDisputes.js]: i always try not to compute inside the return statement, but if you prefer like this is up to you

File: src/GlobalErrorHandler.js (hunk 2 above)
Comment (fabriziovigevani) [reply src/GlobalErrorHandler.js]: I think that since it's a global error and it's not a component that will be a rendered as a child of the main `Disputes` component it may make more sense to leave it in the generic `errors` folder what do you think ?

Comment (rperez89) [reply src/GlobalErrorHandler.js]: mmm but is not a global error for the whole app, is an error only for disputes so i think that should be in the ddispute folder

Comment (fabriziovigevani) [reply src/GlobalErrorHandler.js]: Yeah okay that's also true.

File: src/components/Disputes/DisputeNotFoundError.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+      </h1>
+      <div
+        css={`
+          margin-bottom: ${3 * GU}px;
+          text-align: center;
+          color: ${theme.surfaceContentSecondary};
+          ${textStyle('body2')};
+        `}
+      >
+        It looks like there’s no dispute associated with the “DisputeID #
```
Comment (sohkai) [inline src/components/Disputes/DisputeNotFoundError.js]: Let's remove the `the` here

File: src/hooks/useDisputes.js [hunk 6]
```diff
@@ -55,10 +55,35 @@ export default function useDisputes() {
...
+  const disputeProcessed = useProcessedDispute(dispute)
+
+  const disputePhase = getPhaseAndTransition(dispute, courtConfig, now)
+  const disputePhaseKey = disputePhase
+    ? convertToString(Object.values(disputePhase)[0])
+    : ''
+
+  return useMemo(() => {
+    if (fetching || (dispute && !disputeProcessed)) {
+      return { fetching: true }
```
Comment (sohkai) [inline src/hooks/useDisputes.js]: Not sure how the rest of the project's hooks have been standardized on, but in this case, it might be nice to return an array instead, rather than two different type schemas.

So for example:

```
return [null, true] // fetching

return [{ ... }, false] // has information
```

Comment (fabriziovigevani) [reply src/hooks/useDisputes.js]: When the hook returns lots of different data we usually just use an object.

In this case i think your suggestion makes sense.

> But if all the other code is structured like how this hook is currently written, with an implicitly false fetching key when data exists

It's not, we can change it without extra effort
